esmvaltool/recipes/recipe_collins13ipcc.yml

```diff
@@ -0,0 +1,3032 @@
+        # IAV_calc_clt and set iavmode to "each" there too)
+        # iavmode:
+        # Show difference in percent (1, default 0)
+        # percent:
+
+      ch12_plot_map_diff_mmm_stipp_clt:
+        <<: *clt_map_settings
+        script: ipcc_ar5/ch12_plot_map_diff_mmm_stipp.ncl
+
+        ### Required attributes
+        ancestors: ['clt', 'ch12_calc_map_diff_mmm_clt']
+
+        ### Optional attributes
+        # Map projection (any valid ncl projection, default Robinson)
+        # projection: Robinson
+        title: "Annula mean cloud fraction change"
+        # Contour levels for all difference plots
+        diff_levs: [-10, -9, -8, -7, -6, -5, -4, -3, -2, -1, 0,
+                    1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        # Maximum number of plots in vertical
+        max_vert: 2
+        # Maximum number of plots in horizontal
+        max_hori: 2
+        # Alternative colormap (path to .rgb file or ncl name)
+        colormap: $diag_scripts/shared/plot/rgb/ipcc_colors_red2blu_centered.rgb
+        # Span whole colormap (default false)
+        # span:
+        # Plot significance as stippling
+        sig: true
+        # Plot uncertainty as hatching
+        not_sig: true
+        # Alternative name for output plot
+        # (default diagnostic + varname + time_avg)
+        # pltname:
+        # Print number of models at top-right of the figure
+        model_nr: true
+
+  ###  Mean sea level pressure change in RCPs for multi model mean ############
+  IAV_calc_psl:
+    description: Calculate sea level pressure interannual variability
+                 for stippling significance
+    themes:
+      - varmodes
+    realms:
+      - atmos
+    variables:
+      psl:
+        preprocessor: preproc_map
+        project: CMIP5
+        mip: Amon
+        exp: piControl
+        ensemble: r1i1p1
+    additional_datasets:
+
+      - {dataset: CanESM2,       start_year: 2015, end_year: 3010}
+      - {dataset: CESM1-CAM5,    start_year:    1, end_year:  319}
+      - {dataset: CSIRO-Mk3-6-0, start_year:    1, end_year:  500}
+      - {dataset: GFDL-ESM2G,    start_year:    1, end_year:  500}
+      - {dataset: IPSL-CM5A-LR,  start_year: 1800, end_year: 2799}
+      - {dataset: MPI-ESM-LR,    start_year: 1850, end_year: 2849}
+
+    scripts:
+      ch12_calc_IAV_for_stippandhatch_psl: &IAV_settings_psl
+        script: ipcc_ar5/ch12_calc_IAV_for_stippandhatch.ncl
+
+        ### Required attributes
+        # Plot style
+        styleset: CMIP5
+        # Time average ('opt' argument of time_operations.ncl,
+        # annualclim, seasonalclim)
+        time_avg: seasonalclim
+
+        ### Optional attributes
+        # Length of period in years to calculate variability over,
+        # default total time period
+        periodlength: 20.
+        # Calculate IAV from multi-model mean (mmm, default) or
+        # save individual models (each)
+        # iavmode:
+
+  psl_ch
```

Comment:
This syntax is no longer supported since ESMValGroup/ESMValCore#170

Reply:
Yes I am just working on the changes related to ESMValGroup/ESMValCore#170

The mask_landseaice works fine (and as far as I understand all I need to change is remove the fx_files: [sftlf, sftgif] in the recipe, no changes in diagnostic, correct?)

What I do not understand though is how the very last diagnostic in the recipe needs to change (L2930 ff).
Thats where I am at:

    variables:
      snw:
        project: CMIP5
        mip: LImon
        ensemble: r1i1p1
        exp: historical
      sftlf:
        project: CMIP5
        mip: fx
        exp: historical
        ensemble: r0i0p0
      sftgif:
        project: CMIP5
        mip: fx
        exp: historical
        ensemble: r0i0p0

The files that exist have names like sftlf_fx_CanESM2_historical_r0i0p0.nc but the preprocessor looks for something sftlf_fx_CanESM2_historical_r0i0p0_*.nc, so somehow adding the _ where the years would follow which I use for snw. It also complained without the project, exp, ensemble, so I have to add 10 lines (which I do not think was the goal of this was it?).
Is there an example somewhere I could use to check?

Reply:
The `fx` variables are now defined and read as standard variables.
So you probably need to adapt your diag script to read multiple variables.

Regarding the filename I don't know how this changed, @valeriupredoi can you help?